src/indexing.jl

```diff
@@ -369,4 +369,14 @@ Base.checkindex(B::Type{Bool}, inds::AbstractUnitRange, i::StaticIndexing{T}) wh
 
 # unsafe_view
 
-Base.unsafe_view(A::AbstractArray, i::StaticIndexing{T}) where T = Base.unsafe_view(A, unwrap(i))
+# I was considering catching arbitrary `Union{StaticIndexing,Base.ViewIndex}`
+# indices here but they shouldn't ever be mixed here.
+# i1 is explicitly specified to avoid ambiguities with Base
```

Comment:
I like these comments a lot: the indexing code is quite tricky and it's good to have some context for why things are like they are.

One minor thing - you could remove the reference to `I` here. (The fact that it's you who's thought carefully about this is definitely relevant to the discussion on this PR, but will be somewhat irrelevant to people looking at the code later.) Perhaps could be written:

```julia
# unsafe_view need only deal with vargs of `StaticIndexing`, as wrapped by to_indices.
# i1 is explicitly specified to avoid ambiguities with Base
```